src/stores/auth/store.ts

```diff
@@ -0,0 +1,135 @@
+import { create } from "zustand"
+import { refreshTokens, serverLogout } from "./api"
+import { clearAuthTimers, scheduleAuthTimers } from "./timers"
+import type { Session, TokenResponse } from "./types"
+
+const TEN_HOURS_MS = 10 * 60 * 60 * 1000
+const SESSION_START_AT_KEY = "auth.sessionStartAt"
+const ID_TOKEN_KEY = "auth.idToken"
+
+const getSessionStartAt = (): number | null => {
+  const raw = typeof window !== "undefined" ? localStorage.getItem(SESSION_START_AT_KEY) : null
+  return raw ? Number(raw) : null
+}
+
+const setSessionStartAt = (ts: number | null): void => {
+  if (typeof window === "undefined") return
+  if (ts) localStorage.setItem(SESSION_START_AT_KEY, String(ts))
+  else localStorage.removeItem(SESSION_START_AT_KEY)
+}
+
+const getIdToken = (): string | undefined =>
+  typeof window !== "undefined" ? (localStorage.getItem(ID_TOKEN_KEY) ?? undefined) : undefined
+
+const setIdToken = (v?: string): void => {
+  if (typeof window === "undefined") return
+  if (v) localStorage.setItem(ID_TOKEN_KEY, v)
+  else localStorage.removeItem(ID_TOKEN_KEY)
+}
+
+type AuthStore = {
+  session: Session | null
+  showExpiryWarning: boolean
+  isRefreshing: boolean
+  loginFromTokens: (t: TokenResponse, opts?: { resetSessionWindow?: boolean }) => void
+  refresh: () => Promise<boolean>
+  logout: (reason?: "manual" | "expired") => Promise<void>
+  setShowExpiryWarning: (v: boolean) => void
+  bootstrap: () => Promise<void>
+}
+
+export const useAuthStore = create<AuthStore>((set, get) => ({
+  session: null,
+  showExpiryWarning: false,
+  isRefreshing: false,
+
+  setShowExpiryWarning: (v) => set({ showExpiryWarning: v }),
+
+  // Called after popup login OR after the user chooses "Stay logged in"
+  loginFromTokens: (t, opts) => {
+    const now = Date.now()
+    // If resetSessionWindow=true (fresh login), start a new 10h window
+    let startAt = getSessionStartAt()
+    if (opts?.resetSessionWindow || !startAt) {
+      startAt = now
+      setSessionStartAt(startAt)
+    }
+    const endsAt = startAt + TEN_HOURS_MS
+
+    // If somehow already past end, bail to logout instead
+    if (now >= endsAt) {
+      void get().logout("expired")
+      return
+    }
+
+    setIdToken(t.id_token)
+
+    set({
+      session: {
+        accessToken: t.access_token,
+        accessExpiresAt: now + t.expires_in * 1000,
+        refreshExpiresAt: now + t.refresh_expires_in * 1000,
+        sessionEndsAt: endsAt,
+        idToken: t.id_token ?? getIdToken(),
+      },
+      showExpiryWarning: false,
+    })
+
+    scheduleAuthTimers({
+      getSession: () => get().session,
+      onRefresh: get().refresh,
```

Comment:
This reference can become stale, use

`onRefresh: () => get().refresh(), // Always get current function`